omod/src/main/webapp/pages/login.gsp

```diff
@@ -36,18 +36,76 @@ ${ ui.includeFragment("referenceapplication", "infoAndErrorMessages") }
             jQuery('#sessionLocationInput').val(jQuery(this).attr("value"));
             updateSelectedOption();
         });
-        
-        jQuery('#login-form').submit(function(e) {
+		jQuery('#sessionLocation li').focus( function() {
+            jQuery('#sessionLocationInput').val(jQuery(this).attr("value"));
+            updateSelectedOption();
+        });
+		
+		jQuery('#username').focus();
+		/* 
+		 * Listen for <ENTER> key press and submit form if pressed
+		 */
+		jQuery('#sessionLocation').keyup(function (e) {
+    		var key = e.which || e.keyCode;
+    		if (key === 13 || updateSelectedOption()) {
+      			jQuery('#login-form').submit();
+				console.log("Submitted");
+    		}
+		});
+		
+		/**
+		 * Get every <li> item in the session location area and create an array from it 
+		 */
+		var buttonValues = Array.from(jQuery('#sessionLocation li'));
+		for (var i = 0; i < buttonValues.length; i++){
+			buttonValues[i].addEventListener('keyup', function (event){
+				var keyCode = event.which || event.keyCode; 
+				switch (keyCode) {
+					// Left arrow key pressed
+					case 37: 
+						jQuery(this).prev('#sessionLocation li').focus();
+						break;
+						
+					// Right arrow key pressed
+					case 39: 
+						jQuery(this).next('#sessionLocation li').focus();
+						break;
+					
+					/*
+					 * For the Arrow Up and Arrow Down key press events, 
+					 * Get the value of the <value> attribute of the currently focused <li> and 
+					 * focus a new locations based on that value
+					*/
+					// Up arrow key pressed
+					case 38:
+						switch (jQuery(this).attr("value")) {
+							case 7: jQuery('#sessionLocation li[id="Inpatient Ward"').focus(); break;
+							case 5: jQuery('#sessionLocation li[id="Laboratory"').focus(); break;
```

Comment:
Here is an example of such locations: http://int02.openmrs.org/openmrs/login.htm